client/src/components/TreeDropdownField/TreeDropdownField.js

```diff
@@ -496,9 +496,14 @@ class TreeDropdownField extends Component {
     const loading = this.props.loading.indexOf(visibleTree.id || 0) > -1;
     const failed = this.props.failed.indexOf(visibleTree.id || 0) > -1;
     const breadcrumbs = this.getBreadcrumbs();
-    const value = (this.props.data.multiple)
-      ? this.props.value
-      : [this.props.value];
+
+    // Always coerce value to array for menu (even if not multi-select)
+    let value = [];
+    if (this.props.value) {
+      value = Array.isArray(this.props.value)
+        ? this.props.value
+        : [this.props.value];
+    }
```

Comment:
Even after the discussion we've had, I don't agree with coercing the value to array is correct for `TreeDropdownMenu`.

It makes more sense to me that the value proptype should be accepting a string or array, rather than coercing to an array value.

I'll push up an update soon to address this.